### PR TITLE
Fixes for symfony 5 + readme for php 8

### PR DESCRIPTION
**Pull request by rabauss**

### CODE REVIEW

File: src/DependencyInjection/Configuration.php [hunk 1]
```diff
@@ -24,8 +24,14 @@ class Configuration implements ConfigurationInterface
...
-        $treeBuilder = new TreeBuilder();
-        $rootNode = $treeBuilder->root('mvo_contao_facebook_import');
+        $treeBuilder = new TreeBuilder('mvo_contao_facebook_import');
+
+        // Keep compatibility with symfony/config < 4.2
+        if (method_exists($treeBuilder, 'getRootNode')) {
+            $rootNode = $treeBuilder->getRootNode();
+        } else {
+            $rootNode = $treeBuilder->root('mvo_contao_facebook_import');
+        }
```
Comment (fritzmg) [inline src/DependencyInjection/Configuration.php]: Since Contao 4.4 reached the end of its bugfix live, it might be easier to just require `symfony/config: ^4.4 || ^5.2`.

Comment (rabauss) [reply src/DependencyInjection/Configuration.php]: You're right, it's kind of philosophy how long @m-vo wants to support Contao 4.4.
I just copied the logic from other extensions which still support Contao 4.4 and Symfony 5 ;-)
Finally, I added all current symfony + doctrine dependencies to `composer.json`, which could be reduced to Symfony `^4.4 || ^5.2` without `^3.3`!

Comment (rabauss) [reply src/DependencyInjection/Configuration.php]: push @m-vo 😁

Comment (m-vo) [reply src/DependencyInjection/Configuration.php]: Ah sorry, this went off my radar for some reason.

Yes, let's please go with Sf `^4.4 || ^5.2` and also contao/core-bundle `^4.9` as lower boundary. May I ask you to please also drop this nasty bit in this go? https://github.com/m-vo/contao-facebook-import/blob/9b86a40a10c593954007fa0d56a55af719620cc5/src/Resources/contao/config/config.php#L38-L49 :smile:

File: composer.json [hunk 2]
```diff
@@ -21,12 +21,21 @@
     "php": ">=7.1",
     "ext-PDO": "*",
     "ext-json": "*",
+    "codefog/contao-haste": "^4.17",
```
Comment (m-vo) [inline composer.json:24]: I'm fine with reordering packages. Let's also add `sort-packages: true` to the `config` section, though.

**Review by m-vo — CHANGES_REQUESTED** [review]

File: src/DependencyInjection/Configuration.php (hunk 1 above)
Comment (rabauss) [reply src/DependencyInjection/Configuration.php]: Done!

File: composer.json (hunk 2 above)
Comment (rabauss) [reply composer.json:24]: Done!

**Comment by rabauss** [comment]
I was in flow - and somehow I did a lot more refactoring :-D

**Comment by m-vo** [comment]
> I was in flow - and somehow I did a lot more refactoring :-D

I really appreciate all the work you put into this but I'm not sure I like all of this. :see_no_evil: 

IMHO this PR should only contain the changes related to PHP8 / Sf. As we talked before it's a sensible choice to raise dependencies and also include the related changes for this. :+1: 

Rewriting the code to use the new + fancy stuff while the old ways of doing it is still supported (like using an event subscriber for loading assets or cron registration with tags…) doesn't make much sense to me. This potentially introduces new bugs without a real benefit. (With good unit testing in place it'd probably be a different story…)

Same goes IMHO for the CS changes. We're moving a lot of files here and changing things just to follow a new (equally arbitrary) convention but are going to loose a coherent and readable git history.

File: composer.json [hunk 3]
```diff
@@ -46,5 +56,14 @@
...
+  },
+  "config": {
+    "sort-packages": true
+  },
+  "repositories": [
+    {
+      "type": "vcs",
+      "url": "https://github.com/brainexe/php-graph-sdk.git"
+    }
+  ]
```
Comment (m-vo) [inline composer.json:68]: Shouldn't we omit this as well as the `dev-php8 as 5.7-dev` and let the user add it in their application `composer.json` like you stated in the readme?

File: src/ContaoManager/Plugin.php [hunk 4]
```diff
@@ -62,7 +62,7 @@ public function getPackageDependencies()
      */
     public function registerContainerConfiguration(LoaderInterface $loader, array $managerConfig)
     {
-        $loader->load(__DIR__.'/../Resources/config/config.yml');
+        $loader->load(__DIR__.'/../../config/config.yaml');
```
Comment (m-vo) [inline src/ContaoManager/Plugin.php]: This change seems unrelated.

File: src/ContaoManager/Plugin.php [hunk 5]
```diff
@@ -72,8 +72,8 @@ public function registerContainerConfiguration(LoaderInterface $loader, array $m
      */
     public function getRouteCollection(LoaderResolverInterface $resolver, KernelInterface $kernel)
     {
-        return $resolver
-            ->resolve(__DIR__.'/../Resources/config/routing.yml')
-            ->load(__DIR__.'/../Resources/config/routing.yml');
+        $file = __DIR__.'/../../config/routes.yaml';
+
+        return $resolver->resolve($file)->load($file);
```
Comment (m-vo) [inline src/ContaoManager/Plugin.php]: This change seems unrelated.

File: src/DataContainer/FacebookNode.php [hunk 6]
```diff
@@ -29,8 +29,8 @@
 
 class FacebookNode implements FrameworkAwareInterface, ContainerAwareInterface
 {
-    use FrameworkAwareTrait;
     use ContainerAwareTrait;
+    use FrameworkAwareTrait;
```
Comment (m-vo) [inline src/DataContainer/FacebookNode.php]: This change seems unrelated.

File: src/DependencyInjection/MvoContaoFacebookImportExtension.php [hunk 7]
```diff
@@ -33,8 +33,8 @@ class MvoContaoFacebookImportExtension extends Extension
      */
     public function load(array $configs, ContainerBuilder $container)
     {
-        $loader = new YamlFileLoader($container, new FileLocator(__DIR__.'/../Resources/config'));
-        $loader->load('services.yml');
+        $loader = new YamlFileLoader($container, new FileLocator(__DIR__.'/../../config'));
+        $loader->load('services.yaml');
```
Comment (m-vo) [inline src/DependencyInjection/MvoContaoFacebookImportExtension.php]: This change seems unrelated.

File: src/MvoContaoFacebookImportBundle.php [hunk 8]
```diff
@@ -21,4 +21,8 @@
  */
 class MvoContaoFacebookImportBundle extends Bundle
 {
+    public function getPath(): string
+    {
+        return \dirname(__DIR__);
+    }
```
Comment (m-vo) [inline src/MvoContaoFacebookImportBundle.php]: This change seems unrelated.

File: src/Synchronization/Scheduler.php [hunk 9]
```diff
@@ -30,7 +30,6 @@
 class Scheduler implements ContainerAwareInterface, FrameworkAwareInterface
 {
     use ContainerAwareTrait;
-
```
Comment (m-vo) [inline src/Synchronization/Scheduler.php:33]: This change seems unrelated.

**Review by m-vo — CHANGES_REQUESTED** [review]
I'm fine in general, see my remarks. Please remove the unrelated changes, though. :pray: 

There are also lots of files where you checked in permission changes. Why is this needed? If not, please remove them as well. :slightly_smiling_face:

File: composer.json (hunk 3 above)
Comment (fritzmg) [reply composer.json:68]: I think this entry is for the dev dependency.

Comment (m-vo) [reply composer.json:68]: Makes sense. :+1:

File: src/ContaoManager/Plugin.php (hunk 4 above)
Comment (rabauss) [reply src/ContaoManager/Plugin.php]: see: https://github.com/m-vo/contao-facebook-import/pull/33

File: src/ContaoManager/Plugin.php (hunk 5 above)
Comment (rabauss) [reply src/ContaoManager/Plugin.php]: see: https://github.com/m-vo/contao-facebook-import/pull/33

File: src/DataContainer/FacebookNode.php (hunk 6 above)
Comment (rabauss) [reply src/DataContainer/FacebookNode.php]: see: https://github.com/m-vo/contao-facebook-import/pull/33

File: src/DependencyInjection/MvoContaoFacebookImportExtension.php (hunk 7 above)
Comment (rabauss) [reply src/DependencyInjection/MvoContaoFacebookImportExtension.php]: see: https://github.com/m-vo/contao-facebook-import/pull/33

File: src/MvoContaoFacebookImportBundle.php (hunk 8 above)
Comment (rabauss) [reply src/MvoContaoFacebookImportBundle.php]: see: https://github.com/m-vo/contao-facebook-import/pull/33

File: src/Synchronization/Scheduler.php (hunk 9 above)
Comment (rabauss) [reply src/Synchronization/Scheduler.php:33]: see: https://github.com/m-vo/contao-facebook-import/pull/33

**Review by m-vo — APPROVED** [review]
LGTM :tada:

**Comment by m-vo** [comment]
Thank you @rabauss!
